src/ansys/simai/core/utils/numerical.py

```diff
@@ -113,3 +113,13 @@ def validate_tolerance_parameter(tolerance: Optional[Number]):
         raise TypeError(f"The tolerance argument must be a number >= 0 (passed {tolerance})")
     if tolerance < 0:
         raise ValueError(f"The tolerance argument must be a number >= 0 (passed {tolerance})")
+
+
+# Recursively go through dict to cast values to float, so that we support NaN and inf
+def cast_values_to_float(data: Any):
+    if isinstance(data, dict):
+        return {k: cast_values_to_float(v) for k, v in data.items()}
+    elif isinstance(data, str) and data.lower() in ["nan", "inf", "-inf"]:
+        return float(data)
+    else:
+        return data
```

Comment:
IMO We don't want to run this function on any random dict (and convert a string that should be kept string into float). This is reserved for places where the value should be float, we can still handle invalid value, but we need to handle it as such.
```suggestion
    try:
        return float(data)
    except ValueError:
        logger.waning(f"received invalid float value: {data!r}")
        return float("nan")
```

Reply:
Hmm I'm fine with that but then I think we have to reraise instead of returning nan.